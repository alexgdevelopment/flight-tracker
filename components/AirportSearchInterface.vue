<template>
  <div>
    <v-text-field
      v-model="searchValue"
      clearable
      label="Airport name"
      autofocus
      class="px-4"
      data-testid="searchField"
      :counter="10"
      :rules="searchFieldRules"
    ></v-text-field>
    <v-row v-if="showSearchButton" class="mt-3" justify="center">
      <v-col cols="auto">
        <v-btn @click="buttonClick">Go</v-btn>
      </v-col>
    </v-row>
    <AirportDetail v-if="currentAirport" />
    <v-col v-else-if="noResultsInSearch">
      <p>No results.</p>
    </v-col>
    <v-col
      v-else-if="airports"
      class="results-container"
      data-testid="airportsContainer"
    >
      <AirportCard
        v-for="airport in airports"
        :key="airport.fullName"
        class="mb-3"
        :airport="airport"
        @click.native="airportClicked(airport)"
      />
    </v-col>
  </div>
</template>

<script>
  import { mapState } from 'vuex'
  export default {
    data() {
      return {
        searchFieldRules: [
          (v) => v === null || v.length > 2 || 'Min 3 characters',
          (v) => v === null || v.length <= 10 || 'Max 10 characters',
        ],
      }
    },
    computed: {
      ...mapState({
        mainColor: (state) => state.mainColor,
        airports: (state) => state.airports.airports,
        currentAirport: (state) => state.airports.currentAirport,
      }),
      noResultsInSearch() {
        return this.airports?.length === 0
      },
      searchValue: {
        get() {
          return this.$store.state.searchStringParam
        },
        set(value) {
          this.$store.commit('setSearchStringParam', { searchStringParam: value })
        },
      },
      showSearchButton() {
        return this.searchValue !== '' && this.searchValue !== null
      },
    },
    methods: {
      buttonClick() {
        this.$store.dispatch('search')
      },
      airportClicked(airport) {
        this.$store.commit('airports/setCurrentAirport', { airport })
      },
    },
  }
</script>