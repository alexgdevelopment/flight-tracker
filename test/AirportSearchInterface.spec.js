import { shallowMount } from '@vue/test-utils'
import { Store } from 'vuex'
import { mockAirport } from './mockObjects'
import AirportSearchInterface from '@/components/AirportSearchInterface.vue'

describe('AirportSearchInterface', () => {
  let wrapper = null
  let store
  let state

  beforeEach(() => {
    state = {
      mainColor: 'green',
    }
    store = new Store({
      state,
      modules: {
        airports: {
          state: {
            airports: [mockAirport],
          },
        },
      },
    })
    wrapper = shallowMount(AirportSearchInterface, {
      store,
      stubs: {
        'v-text-field': {
          template: '<div class="v-text-field" />',
        },
        AirportCard: {
          template: '<div class="airport-card" />',
        },
      },
    })
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
