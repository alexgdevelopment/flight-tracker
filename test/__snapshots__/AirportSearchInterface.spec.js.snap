// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirportSearchInterface should match the snapshot 1`] = `
<div>
  <div
    autofocus="autofocus"
    class="v-text-field px-4"
    clearable=""
    counter="10"
    label="Airport name"
    rules="(v) => {
          /* istanbul ignore next */
          cov_19eqwr82h5().f[2]++;
          cov_19eqwr82h5().s[6]++;
          return (
            /* istanbul ignore next */
            (cov_19eqwr82h5().b[0][0]++, v === null) ||
            /* istanbul ignore next */
            (cov_19eqwr82h5().b[0][1]++, v.length > 2) ||
            /* istanbul ignore next */
            (cov_19eqwr82h5().b[0][2]++, 'Min 3 characters')
          );
        },(v) => {
          /* istanbul ignore next */
          cov_19eqwr82h5().f[3]++;
          cov_19eqwr82h5().s[7]++;
          return (
            /* istanbul ignore next */
            (cov_19eqwr82h5().b[1][0]++, v === null) ||
            /* istanbul ignore next */
            (cov_19eqwr82h5().b[1][1]++, v.length <= 10) ||
            /* istanbul ignore next */
            (cov_19eqwr82h5().b[1][2]++, 'Max 10 characters')
          );
        }"
  />
   
  <v-row-stub
    class="mt-3"
    justify="center"
    tag="div"
  >
    <v-col-stub
      cols="auto"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        tag="button"
        type="button"
      >
        Go
      </v-btn-stub>
    </v-col-stub>
  </v-row-stub>
   
  <v-col-stub
    class="results-container"
    tag="div"
  >
    <div
      airport="[object Object]"
      class="airport-card mb-3"
    />
  </v-col-stub>
</div>
`;
